src/stores/useProductEditorStore.ts

```diff
@@ -0,0 +1,48 @@
+import { create } from 'zustand';
+
+export type ProductData = {
+  images: {
+    id: number;
+    url: string;
+  }[];
+  title: string;
+  categories: { id: number; name: string; isSelected: boolean }[];
+  price: number | null;
+  content: string;
+  myLocation: { id: number; name: string };
+};
+
+type OpenPanelParams =
+  | { mode: 'edit'; data: ProductData; id: number }
+  | { mode: 'add'; data?: never; id?: never };
```

Comment:
never면 타입에서 빼주어도 동작할까요??
```suggestion
  | { mode: 'add'; };
```

Reply:
빼도 동작에는 문제가 없습니다.
하지만 넣은 이유는 `add` 일 경우 data, id를 넣었을 때 ts에서 경고해 주길 원했습니다.